src/ducks/auth.ts

```diff
@@ -1701,5 +1727,58 @@ export const useAuthenticationStore = create<AuthStore>()((set, get) => {
       password: string,
       activeAccountId?: string | null,
     ) => getKeyFromKeyManager(password, activeAccountId),
+
+    importSecretKey: async (secretKey: string, password: string) => {
+      set({ isLoading: true, error: null });
+
+      try {
+        const keypair = Keypair.fromSecret(secretKey);
+        const publicKey = keypair.publicKey();
+        const privateKey = keypair.secret();
+
+        const existingAccounts = await getAllAccounts();
+        const accountExists = existingAccounts.some(
+          (account) => account.publicKey === publicKey,
+        );
+
+        if (accountExists) {
+          throw new Error(t("authStore.error.accountAlreadyExists"));
+        }
+
+        const temporaryStore = await getTemporaryStore();
+        if (!temporaryStore) {
+          throw new Error(t("authStore.error.temporaryStoreNotFound"));
+        }
+
+        const keyPair = {
+          publicKey,
+          privateKey,
+        };
+
+        await storeAccount({
+          mnemonicPhrase: temporaryStore.mnemonicPhrase,
+          password,
+          keyPair,
+          accountNumber: existingAccounts.length + 1,
+          shouldRefreshHashKey: false,
+          shouldSetActiveAccount: true,
+          imported: true,
```

Comment:
```suggestion
          importedFromSecretKey: true,
```

I'm suggesting to make it more descriptive so we don't get confused with the other existing `importWallet` function. Let's make it explicit that this one is being imported from an external secret key so we know it's not being derived from the seed phrase. Does that make sense?